concordium-node/src/network/buckets.rs

```diff
@@ -142,8 +158,17 @@ mod tests {
         };
 
         // and check that only one is inserted
-        buckets.insert_into_bucket(p2p_peer, Default::default());
-        buckets.insert_into_bucket(p2p_duplicate_peer, Default::default());
-        assert_eq!(buckets.buckets.len(), 1);
+        let dummy_gauge = GenericGaugeVec::new(Opts::new("", ""), &[""]);
+        if let Ok(bucket_size_gauge_dummy) = dummy_gauge {
+            buckets.insert_into_bucket(p2p_peer, Default::default(), &bucket_size_gauge_dummy);
+            buckets.insert_into_bucket(
+                p2p_duplicate_peer,
+                Default::default(),
+                &bucket_size_gauge_dummy,
+            );
+            assert_eq!(buckets.buckets.len(), 1);
+        } else {
+            panic!("Unable to create dummy gauge.")
```

Comment:
Is this appropriate?

Reply:
Yes, it is fine to panic in a test like this.
You could avoid the if statement by using `.expect("...")` when creating the dummy gauge

Reply:
Good suggestion, I think it is much more idiomatic (and clearer) to do it like that.